sql/src/test/java/io/crate/operation/aggregation/impl/PercentileAggregationTest.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.aggregation.impl;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.metadata.FunctionIdent;
+import io.crate.operation.aggregation.AggregationTest;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.junit.Test;
+
+public class PercentileAggregationTest extends AggregationTest {
+
+    private Object[][] executeAggregation(DataType dataType, Object[][] data) throws Exception {
+        String name = "percentile";
+        return executeAggregation(name, dataType, data, ImmutableList.of(dataType, DataTypes.DOUBLE));
+    }
+
+    @Test
+    public void testReturnTypes() throws Exception {
+        FunctionIdent synopsis1 = new FunctionIdent("percentile", ImmutableList.<DataType>of(DataTypes.LONG, DataTypes.DOUBLE));
+        assertEquals(DataTypes.DOUBLE, functions.get(synopsis1).info().returnType());
+
+        FunctionIdent synopsis2 = new FunctionIdent("percentile", ImmutableList.<DataType>of(DataTypes.LONG, new ArrayType(DataTypes.DOUBLE)));
+        assertEquals(new ArrayType(DataTypes.DOUBLE), functions.get(synopsis2).info().returnType());
+    }
+
+    @Test
+    public void testInteger() throws Exception {
+        Double[] fractions = {0.5, 0.8};
+        Object[][] result;
+
+        result = executeAggregation(DataTypes.INTEGER, new Object[][]{
+            {1, fractions[0]},
+            {2, fractions[0]},
+            {3, fractions[0]},
+            {4, fractions[0]},
+            {5, fractions[0]},
+            {6, fractions[0]},
+            {7, fractions[0]},
+            {8, fractions[0]},
+            {9, fractions[0]},
+            {10, fractions[0]}
+        });
+
+        assertEquals(5.5, result[0][0]);
```

Comment:
according to the documentation this should be 5
however it seems that the percentiles are weighted?